org-formation/075-security-hub/security-hub-suppress-lambda.yaml

```diff
@@ -0,0 +1,121 @@
+              def __init__(self):
+                  self.finding_identifiers = []
+
+              def record_handler(self, record):
+                  payload = json.loads(record["body"])
+                  logger.info(f"payload {payload}")
+                  finding_identifier = {
+                      "Id": payload["detail"]["findings"][0]["Id"],
+                      "ProductArn": payload["detail"]["findings"][0]["ProductArn"],
+                  }
+                  self.finding_identifiers.append(finding_identifier)
+
+              def get_finding_identifiers(self):
+                  return self.finding_identifiers
+
+          """
+          Uses the Lambda Powertools BatchProcessor in order to build the findings id list.
+          Lambda powertools is used to prevent successfully processed messages being returned to SQS
+          https://awslabs.github.io/aws-lambda-powertools-python/latest/utilities/batch/
+          """
+          @logger.inject_lambda_context(log_event=True)
+          def handler(event, context):
+              records = event["Records"]
+              logger.info(f"records {records}")
+
+              processor = BatchProcessor(event_type=EventType.SQS)
+
+              rh = RecordsHandler()
+              with processor(records, rh.record_handler) as proc:
+                  proc.process()
+
+              finding_identifiers = rh.get_finding_identifiers()
+              logger.info(f"finding_identifiers {finding_identifiers}")
+
+              try:
+                  response = sh_client.batch_update_findings(
+                      FindingIdentifiers=finding_identifiers,
+                      Workflow={"Status": "SUPPRESSED"},
+                  )
+                  for processed_findings in response["ProcessedFindings"]:
+                      logger.info(
+                          f"processed and suppressed id {processed_findings['Id']} productarn {processed_findings['ProductArn']}"
+                      )
+
+                  for unprocessed_findings in response["UnprocessedFindings"]:
+                      logger.error(
+                          f"unprocessed finding id {unprocessed_findings['FindingIdentifier']['Id']} productarn {unprocessed_findings['FindingIdentifier']['ProductArn']} error code {unprocessed_findings['ErrorCode']} error message {unprocessed_findings['ErrorMessage']}"
+                      )
+
+              except boto3exceptions.ClientError as error:
+                  logger.exception("client error")
+                  raise ConnectionError(f"Client error invoking batch update findings {error}")
+              except boto3exceptions.ParamValidationError as error:
+                  logger.exception("invalid parameters")
+                  raise ValueError(f"The parameters you provided are incorrect: {error}")
+
+              return {"statusCode": 200}
+      Role:
+        !ImportValue 'sagebase-securityhub-suppression-infra-SecurityHubBatchUpdateSuppressionRoleArn'
```

Comment:
this depends on `security-hub-suppress-infra.yaml` and `security-hub-suppress-infra.yaml` depends on this lambda which means that you have a circular dependency.  In order to fix this situation you may need to create a  role for this in here.